rholang/src/test/scala/coop/rchain/rholang/interpreter/CostAccountingReducerTest.scala

```diff
@@ -35,27 +36,36 @@ class CostAccountingReducerTest extends FlatSpec with Matchers with TripleEquals
     val substTerm         = term(Expr(GString("1")))
     val termCost          = Chargeable[Par].cost(substTerm)
     val initCost          = Cost(1000)
-    implicit val costAlg: _cost[Coeval] =
-      loggingCost(CostAccounting.of[Coeval](initCost).value, noOpCostLog)
-    val res = Substitute.charge(Coeval.pure(substTerm), Cost(10000)).attempt.value
-    assert(res === Right(substTerm))
-    assert(costAlg.get() === (initCost - Cost(termCost)))
+    (for {
+      cost <- CostAccounting.initialCost[Task](initCost)
+      res <- {
+        implicit val c = cost
+        Substitute.charge(Task.now(substTerm), Cost(10000)).attempt
+      }
+      _         = assert(res === Right(substTerm))
+      finalCost <- cost.get
+      _         = assert(finalCost === (initCost - Cost(termCost)))
+    } yield ()).runSyncUnsafe(5.seconds)
   }
 
   it should "charge for failed substitution" in {
     val term: Expr => Par = expr => Par(bundles = Seq(Bundle(Par(exprs = Seq(expr)))))
     val varTerm           = term(Expr(EVarBody(EVar(Var(FreeVar(0))))))
     val originalTermCost  = Chargeable[Par].cost(varTerm)
     val initCost          = Cost(1000)
-    implicit val costAlg: _cost[Coeval] =
-      loggingCost(CostAccounting.of[Coeval](initCost).value, noOpCostLog)
-
-    val res = Substitute
-      .charge(Coeval.raiseError[Par](new RuntimeException("")), Cost(originalTermCost))
-      .attempt
-      .value
-    assert(res.isLeft)
-    assert(costAlg.get() === (initCost - Cost(originalTermCost)))
+
+    (for {
+      cost <- CostAccounting.initialCost[Task](initCost)
+      res <- {
+        implicit val c = cost
+        Substitute
+          .charge(Task.raiseError[Par](new RuntimeException("")), Cost(originalTermCost))
```

Comment:
Lets at least add a TODO to insert a suitable exception message

Reply:
The message is irrelevant